src/project/projecttypes.h

```diff
@@ -198,24 +203,83 @@ struct ProjectFile {
         return io::filename(path, includingExtension).toQString();
     }
 
+    bool operator ==(const ProjectFilePath& other) const
+    {
+        return path == other.path
+               && displayNameOverride == other.displayNameOverride;
+    }
+
+    bool operator !=(const ProjectFilePath& other) const
+    {
+        return !(*this == other);
+    }
+};
+
+using ProjectFilePathsList = std::vector<ProjectFilePath>;
+
+struct ProjectFileUrl {
```

Comment:
This struct looks like a duplicate of ProjectFilePath. Also, the names of both structs are not perfect. For example, for something like ProjectFilePath you would expect it to be an io::path alias, but it also contains some additional UI information. What do you think about having only one struct (with the old name) with std::variant<path, QUrl> to store the file location?

Reply:
> What do you think about having only one struct (with the old name) with std::variant<path, QUrl> to store the file location?

That doesn't really solve the problem I was having... Initially, I had just `ProjectFile` with a `QUrl` inside it instead of a `io::path_t`. That worked okay-ish, since as long as the `QUrl` is a local file, you can easily get an `io::path_t` from it if you need it.
But for recent files, we _only_ want to work with `io::path_t`, because only local files should be stored in recent files (I believe those URLs from MuseScore.com expire after some time, so it does not make sense to store them). 
We _could_ use `ProjectFile`-with-`QUrl` in `RecentFilesController`, but that would result in a huge amount of isLocalFile assertions and conversions from `QUrl` to `io::path_t`. And if we'd use `std::variant`, the problem with assertions would be exactly the same (although the conversion from `std::variant` to `io::path_t` is probably more performant than from `QUrl`).

Perhaps, it would be a solution to rename the structs to make their purpose clear: `ProjectFilePath` -> `RecentFile` and `ProjectFileUrl` -> `ProjectFile`. What do you think about that?